OutSystems/outsystems-ui
# ROU-4462: Abstraction to focus on the trigger when we close a component (sidebar, notification, etc)

This PR is for create the abstraction to focus on the trigger when we close a component (sidebar, notification, balloon and bottom sheet) 

### What was happening
- When we close the notification, stay focused on the notification element. The element is hidden but the screen reader keeps focused on the element.
- When the sidebar is closed, it does not return focus to the previous focusable element
- When a component starts opening, the focus was set to the body when closed.

### What was done
- Created the FocusManager class to be instantiated in overlay components to manage the focus when closing the component.
   - The FocusManager.storeLastFocusedElement stores the element activeElement reference
   - The FocusManager.returnFocusToElement return the focus to the stored element. If the element does not exist on the page, is not focusable or is disabled, the focus is set to the first focusable element in the page.

### Test Steps
1. Go to a page with a notification that starts opening
2. Check that when the notification is closed, the focus goes to the first focusable element on the page

1. Go to a page with button to open the notification
2. Click on the button.
3. Check that when the notification is closed, the focus returns to the button.

### Checklist
